setup.py

```diff
@@ -146,213 +352,9 @@ def setup_with_extensions(extensions=None):
-            ],
-            include_dirs=['.', 'pygsti/evotypes', np.get_include()],
-            language="c++",
-            extra_link_args=["-std=c++11"]
-        ),
-        Extension(
-            "pygsti.evotypes.stabilizer.termreps",
-            sources=[
-                "pygsti/evotypes/stabilizer/termreps.pyx",
-                "pygsti/evotypes/stabilizer/termcreps.cpp",
-                "pygsti/evotypes/stabilizer/statecreps.cpp",
-                "pygsti/evotypes/stabilizer/opcreps.cpp",
-                "pygsti/evotypes/stabilizer/effectcreps.cpp"
-            ],
-            include_dirs=['.', 'pygsti/evotypes', np.get_include()],
-            language="c++",
-            extra_link_args=["-std=c++11"]
-        ),
-        Extension(
-            "pygsti.forwardsims.mapforwardsim_calc_densitymx",
-            sources=[
-                "pygsti/forwardsims/mapforwardsim_calc_densitymx.pyx",
-                "pygsti/evotypes/densitymx/statecreps.cpp",
-            ],
-            include_dirs=['.', 'pygsti/evotypes', 'pygsti/evotypes/densitymx', np.get_include()],
-            language="c++",
-            extra_link_args=["-std=c++11"]
-        ),
-        Extension(
-            "pygsti.forwardsims.termforwardsim_calc_statevec",
-            sources=[
-                "pygsti/forwardsims/termforwardsim_calc_statevec.pyx",
-                "pygsti/evotypes/statevec/statecreps.cpp",
-                "pygsti/evotypes/basecreps.cpp"
-            ],
-            include_dirs=['.', 'pygsti/evotypes', 'pygsti/evotypes/statevec', np.get_include()],
-            language="c++",
-            extra_link_args=["-std=c++11"]
-        ),
-        Extension(
-            "pygsti.forwardsims.termforwardsim_calc_stabilizer",
-            sources=[
-                "pygsti/forwardsims/termforwardsim_calc_stabilizer.pyx",
-                "pygsti/evotypes/stabilizer/statecreps.cpp",
-                "pygsti/evotypes/basecreps.cpp"
-            ],
-            include_dirs=['.', 'pygsti/evotypes', 'pygsti/evotypes/stabilizer', np.get_include()],
-            language="c++",
-            extra_link_args=["-std=c++11"]
-        ),
-        Extension(
-            "pygsti.circuits.circuitparser.fastcircuitparser",
-            sources=["pygsti/circuits/circuitparser/fastcircuitparser.pyx"],
-            include_dirs=['.', np.get_include()],
-            language="c++",
-            extra_link_args=["-std=c++11"]
-        )
-    ]
-    setup_with_extensions(cythonize(ext_modules, compiler_directives={'language_level': "3"}, exclude_failures=True))
-except ImportError:
-    # Cython or numpy is not available
-    warn("Extensions build tools are not available. Installing without Cython extensions...")
-    setup_with_extensions()
-except SystemExit:
-    # Extension compilation failed
-    warn("Error in extension compilation. Installing without Cython extensions...")
-    setup_with_extensions()
+except SystemExit as e:
```

Comment:
We still have this try-except block so that we can tell the user about the env variables. However, we now re-raise the exception.